augur/align.py

```diff
@@ -51,79 +38,155 @@ def run(args):
     int
         returns 0 for success, 1 for general error
     '''
-    seq_fname = args.sequences
-    ref_name = args.reference_name
-    ref_fname = args.reference_sequence
-    if args.output:
-        output = args.output
-    else:
-        output = "aligment.fasta"
+    temp_files_to_remove = []
 
     try:
-        seqs = SeqIO.to_dict(SeqIO.parse(args.sequences, 'fasta'))
-    except FileNotFoundError:
-        print("\nCannot read sequences -- make sure the file %s exists and contains sequences in fasta format"%seq_fname)
-        return 1
-    except ValueError as error:
-        print("\nERROR: Problem reading in {}:".format(args.sequences))
-        print(error)
-        return 1
+        check_arguments(args)
+        seqs = read_sequences(*args.sequences)
+        existing_aln = read_alignment(args.existing_alignment) if args.existing_alignment else None
+
+        # if we have been given a reference (strain) name, make sure it is present
+        ref_name = args.reference_name
+        if args.reference_name:
+            ensure_reference_strain_present(ref_name, existing_aln, seqs)
 
-    if ref_name and (ref_name not in seqs):
-        print("Specified reference name %s is not in the sequence sample. Will not trim."%ref_name)
-        ref_name = None
-
-    # potentially add the reference seuqnce to the sequences
-    if ref_fname and (not ref_name):
-        if os.path.isfile(ref_fname):
-            try:
-                ref_seq = SeqIO.read(ref_fname, 'genbank' if ref_fname.split('.')[-1] in ['gb', 'genbank'] else 'fasta')
-            except:
-                print("WARNING: Cannot read reference sequence."
-                      "\n\tmake sure the file %s contains one sequence in genbank or fasta format"%ref_fname)
+        # If given an existing alignment, then add the reference sequence to this if desired (and if it is the same length)
+        if existing_aln and args.reference_sequence:
+            existing_aln_fname = args.existing_alignment + ".ref.fasta"
+            ref_seq = read_reference(args.reference_sequence)
+            if len(ref_seq) != existing_aln.get_alignment_length():
+                raise AlignmentError("ERROR: Provided existing alignment ({}bp) is not the same length as the reference sequence ({}bp)".format(alignment.get_alignment_length(), len(ref_seq)))
```

Comment:
this needs to be `existing_aln.get_a...`

Reply:
👍 ca71c4f